scipy/_lib/_util.py

```diff
@@ -37,8 +37,14 @@
 
 if np.lib.NumpyVersion(np.__version__) >= "2.0.0.dev0":
     try:
-        np_long = np.long  # type: ignore[attr-defined]
-        np_ulong = np.ulong  # type: ignore[attr-defined]
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                action="ignore",
+                message=".*In the future `np.long` will be defined as.*",
```

Comment:
If you want to be more pedantic...
```suggestion
                message=r".*In the future `np\.long` will be defined as.*",
```

Thanks!